resources/shaders/lighting.hlsl

```diff
@@ -154,8 +155,20 @@ float4 DoShadowAllLights(float3 wpos, float3 V, float3 normal, float metallic, f
 		float light_dist = length(L);
 		L /= light_dist;
 
+		float dir_dot = dot(L, normal);
+
+		if (dir_dot < 0.0f)
+		{
+			continue;
+		}
+
 		float attenuation = calc_attenuation(light, L, light_dist);
 
+		if (attenuation < 0.2f)
```

Comment:
Potential sharp cut off? light.rad works for spot and point lights as well, but be sure to ignore it if it's a directional light.

Reply:
👍  nice catch